source/GameInterface/DynamicSync/DynamicSyncRegistry.cs

```diff
@@ -0,0 +1,473 @@
+        {
+            if (memberInfo is not FieldInfo && memberInfo is not PropertyInfo)
+                return false;
+
+            if (!registrations.ContainsKey(type))
+            {
+                registrations.Add(type, new DynamicSyncRegistryItem());
+            }
+
+            if (registrations[type].Members.Contains(memberInfo))
+                return false;
+
+            registrations[type].Members.Add(memberInfo);
+
+            return true;
+        }
+
+        public bool AddTargetMethod(Type type, MethodInfo methodInfo)
+        {
+
+            if (!registrations.ContainsKey(type))
+            {
+                registrations.Add(type, new DynamicSyncRegistryItem());
+            }
+
+            if (registrations[type].TargetMethods.Contains(methodInfo))
+                return false;
+
+            registrations[type].TargetMethods.Add(methodInfo);
+
+            return true;
+        }
+
+        // TODO: Find a cleaner way of keeping the assembly and handlers for testing purposes
+        public static Assembly Assembly { get; set; }
+
+        public static IEnumerable<Type> DynamicHandlers  { get; set; }
+
+        public DynamicSyncRegistry(IObjectManager objectManager, DynamicSyncPatchProcessor dynamicSyncPatchProcessor)
+        {
+            this.objectManager = objectManager;
+            this.dynamicSyncPatchProcessor = dynamicSyncPatchProcessor;
+        }
+
+        public void Build()
+        {
+            List<Assembly> assemblies = new List<Assembly>
+            {
+                Assembly.GetExecutingAssembly(),
+            };
+
+            // We need to load different dlls based on the runtime
+            // currently the games runs .netframework 4.7.2
+            // but tests uses .net 6.0
+            if(System.Environment.Version.Major <= 4)
+            {
+                assemblies.Add(typeof(System.Collections.ArrayList).Assembly);
+                assemblies.Add(typeof(System.Runtime.AssemblyTargetedPatchBandAttribute).Assembly);
+                assemblies.Add(typeof(System.Linq.Enumerable).Assembly);
+                assemblies.Add(typeof(Queue<>).Assembly);
+                assemblies.Add(typeof(Console).Assembly);
+            }
+            else
+            {
+                assemblies.Add(typeof(System.Linq.Enumerable).Assembly);
+                assemblies.Add(typeof(Queue<>).Assembly);
+                assemblies.Add(Assembly.GetExecutingAssembly());
+                assemblies.Add(Assembly.Load("System.Runtime"));
+                assemblies.Add(Assembly.Load("System.Private.CoreLib"));
+                assemblies.Add(Assembly.Load("System.Collections"));
+                assemblies.Add(typeof(Console).Assembly);
+
+            }
+            foreach (var assemblyName in Assembly.GetExecutingAssembly().GetReferencedAssemblies())
+            {
+                assemblies.Add(Assembly.Load(assemblyName));
+            }
```

Comment:
This will load an already loaded assembly. Replace with the following instead
```suggestion
            var loadedAssemblies = AppDomain.CurrentDomain.GetAssemblies()
    .ToDictionary(a => a.GetName().FullName);

assemblies.AddRange(Assembly.GetExecutingAssembly()
    .GetReferencedAssemblies()
    .Select(name => 
        {
            if (loadedAssemblies.TryGetValue(name.FullName, out var asm) == false)
            {
                asm = Assembly.Load(name);
                Logger.Information("Loaded assembly {asmName}", asm.FullName);
            } 

            return asm;
        }
    )
);
```